core/spec/models/spree/product/scopes_spec.rb

```diff
@@ -318,4 +318,11 @@
       expect(products_by_store).to contain_exactly(product)
     end
   end
+
+  # Regression test for SD-1439 ambiguous column name: count_on_hand
+  describe '#in_stock.in_stock_or_backorderable' do
```

Comment:
@kgorazd besides unit test we need a request specs with the same parameters as passed in the sentry error that we caught - https://github.com/spree/spree/blob/main/api/spec/requests/spree/api/v2/storefront/products_spec.rb